client/components/AuthForm.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+
+import { useRegisterMutation } from "../features/api";
+
+
+const AuthForm = () => {
+
+    const [register, { isLoading, isError, data }] = useRegisterMutation();
+    const [email, setEmail] = useState("");
```

Comment:
nit: I usually add space between mutation hooks and state hooks, for readability
```suggestion
    const [register, { isLoading, isError, data }] = useRegisterMutation();
    
    const [email, setEmail] = useState("");
```